src/plugins/intel_gpu/src/runtime/ocl/ocl_device.cpp

```diff
@@ -240,6 +261,12 @@ device_info init_device_info(const cl::Device& device) {
 
         info.supports_imad = info.supports_imad || (features & CL_DEVICE_FEATURE_FLAG_DP4A_INTEL);
         info.supports_immad = info.supports_immad || (features & CL_DEVICE_FEATURE_FLAG_DPAS_INTEL);
+        if (info.dev_type == device_type::discrete_gpu ||
+            info.gfx_ver.major > 12 || (info.gfx_ver.major == 12 && info.gfx_ver.minor >= 70)) {
+            info.has_separate_cache = true;
+        }
+        GPU_DEBUG_INFO << "GPU version: " << (int)info.gfx_ver.major << "." << (int)info.gfx_ver.minor << "." << (int)info.gfx_ver.revision <<
```

Comment:
how about replace from (int) to static_cast<int>?

Reply:
applied, thanks!